code/core/src/manager-api/root.tsx

```diff
@@ -337,9 +340,24 @@ export function useStoryPrepared(storyId?: StoryId) {
 
 export function useParameter<S>(parameterKey: string, defaultValue?: S) {
   const api = useStorybookApi();
+  const [parameter, setParameter] = useState(api.getCurrentParameter<S>(parameterKey));
 
-  const result = api.getCurrentParameter<S>(parameterKey);
-  return orDefault<S>(result, defaultValue!);
+  const handleParameterChange = useCallback(() => {
+    const newParameter = api.getCurrentParameter<S>(parameterKey);
+    if (newParameter !== parameter) {
+      setParameter(newParameter);
+    }
```

Comment:
I think the inequality check is unecessary because `useState` setters already optimize for this by not rerendering when the new value is equal to the old.